From 5438d15e79fdbd3c0ad9e156a850d964d257201c Mon Sep 17 00:00:00 2001
From: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
Date: Tue, 22 Aug 2017 20:58:11 +0530
Subject: [PATCH 4/5] cpuinfo: Do not enable cpu node on IBM Power System

On IBM Power System device tree parsing code detects CPU state
and sets enable flag properly. Hence do not call enable() from
cpuinfo code.

Signed-off-by: Vasant Hegde <hegdevasant@linux.vnet.ibm.com>
---
 src/core/cpuinfo.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/core/cpuinfo.cc b/src/core/cpuinfo.cc
index 9628ab3..2cf8eae 100644
--- a/src/core/cpuinfo.cc
+++ b/src/core/cpuinfo.cc
@@ -36,7 +36,9 @@ int n = 0)
   {
     cpu->addHint("icon", string("cpu"));
     cpu->claim(true);                             // claim the cpu and all its children
-    cpu->enable();                                // enable it
+    if (!is_system_ppc_ibm(node))
+      cpu->enable();                                // enable it
+
     return cpu;
   }
 
-- 
1.8.3.1

